Userland/Applications/PixelPaint/GuideTool.cpp

```diff
@@ -110,6 +114,12 @@ void GuideTool::on_mousemove(Layer&, GUI::MouseEvent&, GUI::MouseEvent& image_ev
         relevant_offset = delta.x();
 
     auto new_offset = (float)relevant_offset + m_guide_origin;
+
+    if (m_shift_held && m_snap_size > 0) {
```

Comment:
You can check `image_event.modifiers() == Mod_Shift` here instead of keeping track of it with a member variable. If the user open a dialog with Ctrl+N or Ctrl+Shift+N the `key_up` event will be lost, making it stick in snap mode :)

Reply:
Thank you! I didn't think a MouseEvent would know about (Keyboard-)modifiers...